src/components/ui/WorkExperienceCard.js

```diff
@@ -0,0 +1,110 @@
+// @flow
+
+import React from 'react'
+
+import theme from '../../styles/theme'
+import typo, { typeface } from '../../styles/typo'
+
+const cmz = require('cmz')
+
+const cx = {
+  wrapper: cmz(
+    `
+      background: ${theme.white}
+      border: 1px solid ${theme.lineSilver5}
+      box-shadow: 4px 4px 0px ${theme.baseTuna.fade(0.95)}
+      padding: 24px
+    `
+  ),
+
+  label: cmz(
+    typo.baseText,
+    `
+      margin-top: -6px
+    `
+  ),
+
+  role: cmz(
+    `
+      font-weight: 500
+    `
+  ),
+
+  at: cmz(
+    `
+      font-style: italic
+      margin: 0px 6px
+    `
+  ),
+
+  company: cmz(
+    `
+      font-weight: 400
+    `
+  ),
+
+  dates: cmz(
+    typo.baseText,
+    `
+      font-size: 18px
+      margin-top: 5px
+    `
+  ),
+
+  button: cmz(
+    typeface.extra,
+    `
+      & {
+        line-height: 18px
+        text-transform: uppercase
+        color: ${theme.baseRed}
+        font-size: 13px
+        cursor: pointer
+        margin: 12px 0 -2px 0
+      }
+
+      &:hover {
+        text-decoration: underline
+      }
+    `
+  )
+}
+
+const monthNames = ['January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 'September', 'October', 'November', 'December']
+
+type Props = {
+  id?: number,
+  role?: string | null,
+  company?: string | null,
+  startDate?: Date | null,
+  endDate?: Date | null,
+  returnEditingCardId?: (number) => void
+}
+
+const WorkExperienceCard = (props: Props) => {
+  const { role, company, startDate, endDate } = props
+
+  const formatDate = (date: Date): string => {
+    return date && `${monthNames[date.getMonth()]} ${date.getFullYear()}`
+  }
+
+  const returnEditingCardId = (): void => {
+    if (props.id && props.returnEditingCardId) {
+      props.returnEditingCardId(props.id)
+    }
+  }
```

Comment:
This method name is a bit misleading. And the usage could be a bit different.

The component consumer could use the component like this:

```
const handleEditExperience = id => () => {
  // do stuff with experience with this id
}

// ...

<WorkExperienceCard
  id={431}
  returnEditingCardId={handleEditExperience(431)}
/>
```

Then this part would be simpler:

```
  const returnEditingCardId = (): void => {
    const { returnEditingCardId } = this.props
    returnEditingCardId && returnEditingCardId()
  }
```

This means the UI component doesn't have to care about returning an information that already exists in the parent component. And will just run the function once the use click in the button.

That way the method and prop names could be changed to something more meaningful:

```suggestion
  const handleEditEntry = (): void => {
    const { editEntry } = this.props
    editEntry && editEntry()
  }
```

Before make this update, please make your considerations. Don't need to update right away if you see something dubious with my suggestion, please comment about in this case :)

Reply:
refactored, thanks for the suggestion 😉